From 7c1361bbf15de979cc233f82f29931ad3d99a2eb Mon Sep 17 00:00:00 2001
From: Patrice Chotard <patrice.chotard@foss.st.com>
Date: Thu, 18 Jan 2024 09:29:26 +0100
Subject: [PATCH] mtd: hyperbus: hyperbus-sfdp: Fix hyperbus_s26hx_chip_setup()

n256ks variable is defined as u8, this is not enough in case
of HyperFlash with more than 255 256KB sectors.

Update n256ks variable to u16 to be able to detect S26HSL512
HyperFlash with 256 256KB sectors.

Change-Id: Ie5f45624805b1dd950f9ef5d0719a1d506ff518e
Signed-off-by: Patrice Chotard <patrice.chotard@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/369128
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
---
 drivers/mtd/hyperbus/hyperbus-sfdp.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/mtd/hyperbus/hyperbus-sfdp.c
+++ b/drivers/mtd/hyperbus/hyperbus-sfdp.c
@@ -102,7 +102,8 @@ static int __xipram hyperbus_s26hx_chip_
 					      struct cfi_private *cfi)
 {
 	u16 mfr_id, dev_id1, dev_id2, cfr1v, cfr2v;
-	u8 nregions, nbtm4ks, ntop4ks, n256ks;
+	u8 nregions, nbtm4ks, ntop4ks;
+	u16 n256ks;
 	u32 *erase;
 
 	/* Read manufacturer and Device ID */
